rules/no-unnecessary-polyfills.js

```diff
@@ -0,0 +1,111 @@
+'use strict';
+const semver = require('semver');
+const readPkgUp = require('read-pkg-up');
+const compatTable = require('@babel/preset-env/data/built-ins.json');
+const upperfirst = require('lodash.upperfirst');
+const camelcase = require('lodash.camelcase');
+const getDocsUrl = require('./utils/get-docs-url');
+
+function isRequireCall(node) {
+	return node.callee.name === 'require';
+}
+
+function getVersionFromPkg(cwd) {
+	const pkg = readPkgUp.sync({cwd});
+	return pkg && pkg.pkg.engines && pkg.pkg.engines.node;
+}
+
+const polyfills = {
+	'es6.promise': ['promise-polyfill'],
+	'es7.promise.finally': ['p-finally']
+};
+
+const polyfillMap = Object.keys(compatTable).reduce((current, name) => {
```

Comment:
Just use a `for-of` loop. `reduce()` just makes the code harder to read.

Reply:
or `.map()`